crates/evm/fuzz/src/strategies/state.rs

```diff
@@ -36,12 +37,31 @@ impl EvmFuzzState {
         }
     }
 
-    /// Collects state changes from a [StateChangeset] and logs into an [EvmFuzzState] according to
-    /// the given [FuzzDictionaryConfig].
-    pub fn collect_state_from_call(&self, logs: &[Log], state_changeset: &StateChangeset) {
+    /// Collects call result (if any), state changes from a [StateChangeset] and logs into an
+    /// [EvmFuzzState] according to the given [FuzzDictionaryConfig].
+    pub fn collect_state_from_call(
+        &self,
+        function: &Option<Function>,
+        result: &Bytes,
+        logs: &[Log],
+        state_changeset: &StateChangeset,
+        run_depth: u32,
+    ) {
         let mut dict = self.inner.write();
 
-        // Insert log topics and data.
+        match function {
+            Some(func) => {
+                // Decode result and collect samples to be used in subsequent fuzz runs.
+                if !result.is_empty() {
+                    if let Ok(decoded_result) = func.abi_decode_output(result, true) {
```

Comment:
```suggestion
                    if let Ok(decoded_result) = func.abi_decode_output(result, false) {
```

Reply:
changed in https://github.com/foundry-rs/foundry/pull/7666/commits/d9d8619e7e45077dc7850d47f7609380bd175cc6